sys/linux/socket_netlink_route.txt

```diff
@@ -879,6 +882,114 @@ rtcanmsg {
 
 rtcanmsg_flags = CGW_FLAGS_CAN_ECHO, CGW_FLAGS_CAN_SRC_TSTAMP, CGW_FLAGS_CAN_IIF_TX_OK
 
+lwtunnel_policy [
+	MPLS_IPTUNNEL_DST	nlattr[MPLS_IPTUNNEL_DST, array[mpls_label]]
+	MPLS_IPTUNNEL_TTL	nlattr[MPLS_IPTUNNEL_TTL, int8]
+
+	LWTUNNEL_IP_ID		nlattr[LWTUNNEL_IP_ID, int64be]
+	LWTUNNEL_IP_DST		nlattr[LWTUNNEL_IP_DST, ipv4_addr]
+	LWTUNNEL_IP_SRC		nlattr[LWTUNNEL_IP_SRC, ipv4_addr]
+	LWTUNNEL_IP_TTL		nlattr[LWTUNNEL_IP_TTL, int8]
+	LWTUNNEL_IP_TOS		nlattr[LWTUNNEL_IP_TOS, int8]
+	LWTUNNEL_IP_FLAGS	nlattr[LWTUNNEL_IP_FLAGS, flags[lwtunnel_ip_flags, int16be]]
+	LWTUNNEL_IP_OPTS	nlnest[LWTUNNEL_IP_OPTS, ip_opts_policy]
+
+	ILA_ATTR_LOCATOR	nlattr[ILA_ATTR_LOCATOR, int64]
+	ILA_ATTR_CSUM_MODE	nlnest[ILA_ATTR_CSUM_MODE, ila_attr_csum_mode_policy]
+	ILA_ATTR_IDENT_TYPE	nlnest[ILA_ATTR_IDENT_TYPE, ila_attr_ident_type_policy]
+	ILA_ATTR_HOOK_TYPE	nlnest[ILA_ATTR_HOOK_TYPE, ila_attr_hook_type_policy]
+
+	LWTUNNEL_IP6_ID		nlattr[LWTUNNEL_IP6_ID, int64be]
+	LWTUNNEL_IP6_DST	nlattr[LWTUNNEL_IP6_DST, ipv6_addr]
+	LWTUNNEL_IP6_SRC	nlattr[LWTUNNEL_IP6_SRC, ipv6_addr]
+	LWTUNNEL_IP6_HOPLIMIT	nlattr[LWTUNNEL_IP6_HOPLIMIT, int8]
+	LWTUNNEL_IP6_TC		nlattr[LWTUNNEL_IP6_TC, int8]
+	LWTUNNEL_IP6_FLAGS	nlattr[LWTUNNEL_IP6_FLAGS, flags[lwtunnel_ip_flags, int16be]]
+	LWTUNNEL_IP6_OPTS	nlattr[LWTUNNEL_IP6_OPTS, ip_opts_policy]
+
+# TODO: disable this comment and the header include one, after kernel adds
+# an include for ipv6hdr in the same file
+#	SEG6_IPTUNNEL_SRH	nlattr[SEG6_IPTUNNEL_SRH, ipv6_sr_hdr]
```

Comment:
This attribute has type `struct seg6_iptunnel_encap`.

Reply:
How would the second element of the `seg6_iptunnel_encap` structure would translate into a description?
AFAIU it is an array of  `ipv6_sr_hdr` structs with 0 elements.

Reply:
`[0]` is used to decode variable sized arrays. Looking at the parsing code in `seg6_build_state`, using `struct seg6_iptunnel_encap` is actually also wrong. There is also supposed to be `ipv6addr` at the end and myabe a variable number of `ipv6_sr_hdr`s in the middle. Please see what `seg6_build_state` does.

Reply:
I did some changes.
PTAL.